### PR TITLE
Update envtest

### PR DESCRIPTION
**Pull request by lobziik**
Replace fetch_ext_bins.sh script with vendored setup-envtest tool.

Follow up for #1070

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -45,6 +45,11 @@ else
   IMAGE_BUILD_CMD = $(ENGINE) build
 endif
 
+PROJECT_DIR := $(shell dirname $(abspath $(lastword $(MAKEFILE_LIST))))
+ENVTEST = go run ${PROJECT_DIR}/vendor/sigs.k8s.io/controller-runtime/tools/setup-envtest
+# ENVTEST_K8S_VERSION refers to the version of kubebuilder assets to be downloaded by envtest binary.
+ENVTEST_K8S_VERSION = 1.25
```
Comment (damdo) [inline Makefile:51]: I'd put this at the top so it's easier to find and harder to forget bumping.

File: Makefile [hunk 2]
```diff
@@ -93,7 +98,7 @@ test: ## Run tests
 	$(DOCKER_CMD) hack/ci-test.sh
 
 unit:
-	$(DOCKER_CMD) go test ./pkg/... ./cmd/...
+	$(DOCKER_CMD) KUBEBUILDER_ASSETS="$(shell $(ENVTEST) use $(ENVTEST_K8S_VERSION) -p path --bin-dir $(PROJECT_DIR)/bin)" go test $(GOTEST_FLAGS) ./pkg/... ./cmd/...
```
Comment (damdo) [inline Makefile:101]: I'd probably move the `go test ...` command into its own script so we can be more consistent with the approach we are taking in the control-plane-machine-set-operator repo:
See:
- https://github.com/openshift/cluster-control-plane-machine-set-operator/blob/dad8c9b9bd1e6df0a2f72a3baaeac03f111153bb/hack/test.sh
- https://github.com/openshift/cluster-control-plane-machine-set-operator/blob/dad8c9b9bd1e6df0a2f72a3baaeac03f111153bb/Makefile#L96-L98

and so we can later evolve it from there (e.g. adding the ginkgo test runner (with richer output and tools) and so on). WDYT?

**Review by damdo — COMMENTED** [review]
A couple of minor suggestions but

/lgtm

File: Makefile (hunk 2 above)
Comment (lobziik) [reply Makefile:101]: Sounds good to me, but i would rather do this in follow up pr, along with another improvements, like ginkgo update

File: Makefile (hunk 1 above)
Comment (lobziik) [reply Makefile:51]: With @JoelSpeed we were discussing adding an extra test for checking evntest version and its accordance with another k8s.io dependencies. Would prefer to solve it this way.

File: Makefile (hunk 2 above)
Comment (damdo) [reply Makefile:101]: 👍

File: Makefile (hunk 1 above)
Comment (damdo) [reply Makefile:51]: 👍

**Comment by JoelSpeed** [comment]
/override ci/prow/e2e-vsphere-serial

This PR doesn't affect our E2E tests

**Pull request merged**